Network: Add load balancer pools (#18179)
canonical/lxd: doc/metadata.txt

```

```{config:option} target_backend network-load-balancer-load-balancer-port-properties
:required: "no"

> :shortdesc: "Backend name or names to forward to"

Reviewer: ```suggestion
:shortdesc: "Backend name(s) to forward to"
```

Author: Fixed.

:type: "backend list"
Either a list of backends or a pool is required.
```

```{config:option} target_pool network-load-balancer-load-balancer-port-properties
:required: "no"
:shortdesc: "Pool of instances to forward to"
:type: "string"

> Either a list of backends or a pool is required.

Reviewer: The sentence "Either a list of backends or a pool is required" can sound as if a list of backends or a pool must be provided for `target_backend`. (Same for `target_pool`.) Can a pool be provided to `target_backend`, or does it need to be provided to `target_pool`?

If what you mean is that a value must be provided for *either* `target_backend` or `target_pool`, how about:

````suggestion
If you do not provide a list of backends, then you must provide a pool to `target_pool`.
```

```{config:option} target_pool network-load-balancer-load-balancer-port-properties
:required: "no"
:shortdesc: "Pool of instances to forward to"
:type: "string"
If you do not provide a pool, then you must provide a list of backends to `target_backend`.
````

Author: Fixed.

```

<!-- config group network-load-balancer-load-balancer-properties end -->
<!-- config group network-load-balancer-pool-properties start -->
```{config:option} healthcheck network-load-balancer-pool-properties
:defaultdesc: "`true`"
:required: "no"
:shortdesc: "Whether to enable or disable health checks"
:type: "bool"

```

```{config:option} healthcheck.failure_count network-load-balancer-pool-properties
:defaultdesc: "`1`"
:required: "no"

> :shortdesc: "Failed probe attempts after which an instance is considered unhealthy."

Reviewer: How about adding "Number" to help clarify what this means when skimming the properties table?

```suggestion
:shortdesc: "Number of failed probe attempts after which an instance is considered unhealthy."
```

Author: Fixed.

:type: "integer"

```

```{config:option} healthcheck.interval network-load-balancer-pool-properties
:defaultdesc: "`5`"
:required: "no"

> :shortdesc: "Interval in which to probe the pool's instances."

Reviewer: What is the interval unit? Seconds, minutes, etc.?

Author: Added info about seconds.

:type: "integer"

```

```{config:option} healthcheck.success_count network-load-balancer-pool-properties
:defaultdesc: "`1`"
:required: "no"

> :shortdesc: "Successful probe attempts after which an instance is considered healthy."

Reviewer: ```suggestion
:shortdesc: "Number of successful probe attempts after which an instance is considered healthy."
```

Author: Fixed.

:type: "integer"

```

```{config:option} healthcheck.timeout network-load-balancer-pool-properties
:defaultdesc: "`3`"
:required: "no"

> :shortdesc: "Timeout after a probe appears to be faulty."

Reviewer: What is the timeout unit? Seconds, minutes, etc.?

Author: Added info about seconds.

:type: "integer"

```

```{config:option} protocol network-load-balancer-pool-properties
:defaultdesc: "`tcp`"
:required: "no"
:shortdesc: "Protocol used for ingress pool traffic."
:type: "string"

> Can be either tcp or udp.

Reviewer: Format this as code, since these are the options that would be provided? (And I think the acronyms would be capitalized, TCP and UDP.)

```suggestion
Can be either `tcp` or `udp`.
```

Author: Good call, but I'll keep them lower case as the actual config keys are also written in lower case. The same applies when creating load balancer ports using `lxc network load-balancer port add <net> <addr> (tcp|udp) ...`

Reviewer: Sorry, I wasn't clear --- I meant that if the options weren't code-formatted, then it would make sense to have them uppercase as acronyms. I agree that lowercase, code-formatted `tcp` and `udp` would be best here.
